java/src/jmri/jmrit/dispatcher/AutoActiveTrain.java

```diff
@@ -708,7 +708,7 @@ protected synchronized void setupNewCurrentSignal(AllocatedSection as, boolean f
             if (as == null) {
                 as = _currentAllocatedSection;
             }else{
-               _currentAllocatedSection = as;         //added bill fitch as set speed by signal crashing will null _currentAllocatedSection 
+               _currentAllocatedSection = as;         //added bill fitch to stop crashes in 'set speed by signal' with null _currentAllocatedSection 
```

Comment:
This is updaeting the currentallocatedsection, this routine must not do that, current section is set by initial allocation and then only when a train moves into it.

Reply:
The program was crashing when I started a train from outside the transit and the dispatch failed to create. JMRI crashed with an error. I found that the current allocated section was not set. I did this botch and now dispatcher works.

Reply:
Not sure how to resolve this. I need this edit, or something else that resolves the problem and stops dispatcher crashing. If this edit is unacceptable,I have not got the skill or knowledge of Dispatcher to produce another edit in dispatcher to solve the problem. I can send a panel and instructions to generate the exception again (might take some time to set up again but I should be able to do it) so someone can produce an acceptable edit, but at least we have a clue what is causing the problem because this edit solved it.